### PR TITLE
Add attribute to control login.defs PASS_WARN_AGE

### PR DESCRIPTION
**Pull request by ncs-alane**
Adds `node['os-hardening']['auth']['pass_warn_age']` with a default of `7`, and passes it to the `/etc/login.defs` template.

### CODE REVIEW

**Comment by atomic111** [comment]
@ncs-alane thanks for the nice pr. i need some time to review it. can you please add a short description about the new attribute in the README.md

**Comment by ncs-alane** [comment]
@atomic111 sure thing!

File: spec/recipes/default_spec.rb [hunk 1]
```diff
@@ -35,15 +35,17 @@
...
   it 'default should include os-hardening recipes by default' do
-    expect(chef_run).to include_recipe 'os-hardening::packages'
-    expect(chef_run).to include_recipe 'os-hardening::limits'
-    expect(chef_run).to include_recipe 'os-hardening::login_defs'
-    expect(chef_run).to include_recipe 'os-hardening::minimize_access'
-    expect(chef_run).to include_recipe 'os-hardening::pam'
-    expect(chef_run).to include_recipe 'os-hardening::profile'
-    expect(chef_run).to include_recipe 'os-hardening::securetty'
-    expect(chef_run).to include_recipe 'os-hardening::sysctl'
+    is_expected.to include_recipe 'os-hardening::packages'
```
Comment (artem-sidorenko) [inline spec/recipes/default_spec.rb]: @chris-rock as I remember we have everywhere the `expect(chef_run)` syntax. What do you think, is this change Ok for us?

I personally would like to have this changes everywhere (incl. other chef repos) and in the dedicated PR(s).

**Comment by ncs-alane** [comment]
@artem-sidorenko okay, I've reverted the test changes and moved them to #136.

**Comment by artem-sidorenko** [comment]
@ncs-alane that looks much more better :-) Many thanks for splitting of this PR. May I ask you to  cleanup the commit history by [squashing the commits](https://ariejan.net/2011/07/05/git-squash-your-latests-commits-into-one/)?

@atomic111 LGTM when the history is clean

**Comment by ncs-alane** [comment]
@artem-sidorenko done like dinner!

**Comment by artem-sidorenko** [comment]
@ncs-alane thank you! Lets wait a bit for @atomic111 to have look (it looks like he is a bit busy currently with something). I'll try to review the test improvements in the next days

**Comment by atomic111** [comment]
@ncs-alane thank you for the great work!!!

**Comment by ncs-alane** [comment]
@atomic111 you're very welcome!

**Pull request merged**